### DISCUSSION TITLE
How to integrate in existing Ansible configuration repository?

### DISCUSSION THREAD

**Discussion by mueller-ma** [autobase-tech/autobase#707] [Q&A] [answered]
Hi,
I'd like to integrate this role in my already existing Ansible configuration. I created a host group (`db`), added this role to requirements.txt and added variables as group variables.
However using the following playbook doesn't execute the role:

````
- name: Deploy Postgres
  hosts: db
  roles:
    - role: vitabaks.postgresql_cluster
 ````

````
$ ansible-playbook playbooks/role_postgres.yml

PLAY [Deploy Postgres] ***************************************************************************************

TASK [Gathering Facts] ***************************************************************************************
ok: [db-01]

PLAY RECAP ***************************************************************************************************
db-01                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
````

I saw https://github.com/vitabaks/postgresql_cluster/discussions/677, but IMO it's too dangerous as your playbooks use `hosts: all` and when someone forgets `--limit db`, Ansible installs Postgres on all servers.

**Comment by vitabaks** [comment, marked as answer]
@mueller-ma Hi

`postgresql_cluster` is not a role; it is a project based on Ansible, consisting of many playbooks and about 40 roles.

Please use it according to the README.

**Reply by mueller-ma** [reply]
That means I have to duplicate my Ansible inventory or split it up into two: One for most servers, one for Postgres.

Will it be possible to use such playbook from my first post when this repo has been converted to a collection (see https://github.com/vitabaks/postgresql_cluster/discussions/272)?

**Reply by vitabaks** [reply]
Your inventory file must be compatible with the format that postgresql_cluster assumes.
https://github.com/vitabaks/postgresql_cluster/blob/master/inventory

**Reply by vitabaks** [reply]
> Will it be possible to use such playbook from my first post when this repo has been converted to a collection

So far, I have no plans to change the format to the collection. 

Version 2.0 will be released soon, which will provide a [UI](https://github.com/vitabaks/postgresql_cluster/issues/589) to hide all this "complexity" associated with inventory and variables files.
